controllers/tasks.js

```diff
@@ -144,10 +153,24 @@ const updateTaskStatus = async (req, res) => {
     const { id: userId } = req.userData;
     const task = await tasks.fetchSelfTask(taskId, userId);
 
-    if (task.taskNotFound) return res.boom.notFound("Task doesn't exist");
-    if (task.notAssignedToYou) return res.boom.forbidden("This task is not assigned to you");
-    if (task.taskData.status === "VERIFIED")
+    if (task.taskNotFound) {
+      logger.error(`Error while updating taskId ${req.params.id}: Invalid taskId passed`);
+      return res.boom.notFound("Task doesn't exist");
+    }
+    if (task.notAssignedToYou) {
+      logger.error(`Error while updating taskId ${req.params.id}: Forbidden user`);
+      return res.boom.forbidden("This task is not assigned to you");
+    }
+    if (task.taskData.status === "VERIFIED") {
+      logger.error(`Error while updating taskId ${req.params.id}: Already verified task`);
       return res.boom.forbidden("Status cannot be updated. Please contact admin.");
+    }
+
+    const taskData = await toFirestoreData(req.body);
```

Comment:
@shubham-y can we have a separate function for lines 169 to 173, if it's going to be used frequently in the future, as of now we are using it 3 times. What do you think?

Reply:
I think separating the logic for logging and returning a response in a different function other than the controller function is not recommended